frontend/src/components/SignIn/SignIn.js

```diff
@@ -31,23 +34,32 @@ class SignIn extends Component {
       .then((response) => {
         sessionStorage.jwt = response.headers.authorization;
         changeAuthorizedState(response.data.id);
+
+        this.setState({
+          flags: false,
+        });
       })
       .catch((error) => {
         if (error.message.includes('401')) {
-          const parentForm = document.getElementsByTagName('form')[1];
-          const errorDiv = document.createElement('p');
-          errorDiv.setAttribute('class', 'errorMessage');
-          errorDiv.innerHTML = 'Unable to log in. Check your email and password and try again.';
-          parentForm.insertAdjacentElement('beforeend', errorDiv);
+          this.setState({
+            flags: true,
```

Comment:
It doesn't look like `flags` is being used anywhere. Can you make this work by only using `err`?